varnish-macros/src/stats.rs

```diff
@@ -0,0 +1,162 @@
+use crate::parser_utils::parse_doc_str;
+use serde::Serialize;
+use syn::{punctuated::Punctuated, token::Comma, Data, Field, Fields, Type};
+
+type FieldList = Punctuated<Field, Comma>;
+
+#[derive(Serialize, Clone)]
+struct VscMetricDef {
+    pub name: String,
+    #[serde(rename = "type")]
+    pub counter_type: String, // "counter", "gauge"
+    pub ctype: String,    // "uint64_t" is only option right now
+    pub level: String,    // "info", "debug", etc
+    pub oneliner: String, // "Counts the number of X", etc
+    pub format: String,   // "integer", "bytes", "duration", "bitmap", etc
+    pub docs: String,
+    pub index: Option<usize>,
+}
+
+#[derive(Serialize)]
+struct VscMetadata {
+    version: String,
```

Comment:
If its always a constant, no need to allocate

```suggestion
    version: &'static str,
```